.github/actions/action-setup_task-installPyProject/action.yml

```diff
@@ -0,0 +1,71 @@
+---
+# Adapted from @pvandyken's composite action
+name: Build and cache Python project
+description: Install project dependencies and optionally install project
+inputs:
+  python-version:
+    description: Python version to setup
+    required: true
+    type: string
+  reference:
+    description: Reference ID to define cache
+    required: true
+    type: string
+  manager:
+    description: Dependency manager used (e.g. poetry)
+    required: false
+    default: "poetry"
+    type: string
+  install-library:
+    description: Indicate whether project library should be installed
+    required: false
+    default: false
+    type: boolean
+
+runs:
+  using: composite
+  steps:
+    - name: Clone repo
+      uses: actions/checkout@v3
+
+    # Setup & configure dependency manager
+    - name: Cache manager installation
+      uses: actions/cache@v3
+      id: cached-manager
+      with:
+        path: ~/.local
+        key: ${{ runner.os }}-${{ inputs.manager }}-${{ inputs.reference }}
+
+    - name: Install poetry
+      uses: snok/install-poetry@v1
+      if: ${{ inputs.manager }} == "poetry" && ${{ steps.cached-manager.outputs.cache-hit }} != 'true'
+      id: setup-poetry
+      with:
+        virtualenvs-create: true
+        virtualenvs-in-project: true
+
+    # Setup & configure Python
+    - name: Setup python
+      uses: actions/setup-python@v4
+      id: setup-python
+      with:
+        python-version: ${{ inputs.python-version }}
+        cache: "${{ inputs.manager }}"
+
+    - name: Store pip cache
+      uses: actions/cache@v3
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os}}-pip-${{ steps.setup-python.outputs.python-version }}
+        restore-keys: ${{ runner.os }}-pip-${{ steps.setup-python.outputs.python-version }}
+
+    # Install dependencies with manager
+    - name: Install dependencies
+      if: ${{ steps.setup-python.outputs.cache-hit }} != 'true' && ${{ steps.setup-poetry.outcome }} == 'success'
```

Comment:
What's the purpose of `${{ steps.setup-poetry.outcome }} == 'success'`? If setup-poetry fails, shouldn't the whole workflow fail?

Reply:
This is more future proofing than anything. Right now the workflow only works for poetry so your assessment of it failing is correct, but if this workflow ends up supporting `hatch` in the future the `setup-poetry` step would be skipped. In this situation, `outcome != success`, but the workflow wouldn't fail. But maybe we end up deciding we actually want separate workflows for setting up different dependency managers.

Reply:
Gotcha, no I think a simple workflow makes sense